File: java-examples/src/com/szymczak/task3/Problem3.java
package com.szymczak.task3;

import static com.szymczak.utils.Utils.isPrime;

public class Problem3 {
    public static void main(String[] args) {
        System.out.printf("Largest biggest prime is: %d", getLargestPrimeFactor(600851475143L));
    }

    private static long getLargestPrimeFactor(long n) {
        long biggestPrime = -1;
        long primeToDive = 2;
        while (primeToDive <=n) {
            if (n % primeToDive == 0) {
                n /= primeToDive;
                if (primeToDive > biggestPrime) {
                    biggestPrime = primeToDive;
                }
                primeToDive = 2;
            } else {
                while (true) {
                    if (isPrime(++primeToDive)) break;
                }
            }
        }
        return biggestPrime;
    }
}

File: python_examples/task3/problem_3.py
from utils.utils import is_prime

"""
The prime factors of 13195 are 5, 7, 13 and 29.
What is the largest prime factor of the number 600851475143 ?
"""


def get_biggest_prime_factor(n: int) -> int:
    biggest_prime = -1
    prime_to_divide = 2
    while prime_to_divide <= n:
        if n % prime_to_divide == 0:
            n //= prime_to_divide
            biggest_prime = biggest_prime if biggest_prime > prime_to_divide else prime_to_divide
            prime_to_divide = 2
        else:
            while True:
                prime_to_divide += 1
                if is_prime(prime_to_divide):
                    break
    return biggest_prime


print(f'Biggest prime number is {get_biggest_prime_factor(600851475143)}')
